servicex/query.py

```diff
@@ -354,10 +371,28 @@ async def transform_status_listener(
                     completed=self.current_status.files_completed,
                 )
 
-            if self.current_status.status == Status.complete:
+            if self.current_status.status in (Status.complete, Status.canceled, Status.fatal):
                 self.files_completed = self.current_status.files_completed
                 self.files_failed = self.current_status.files_failed
-                return
+                titlestr = (f'"{self.current_status.title}" '
+                            if self.current_status.title is not None else '')
+                if self.current_status.status == Status.complete:
+                    return
+                elif self.current_status.status == Status.canceled:
+                    logger.warning(
+                        f"Request {titlestr}canceled: "
+                        f"{self.current_status.files_completed}/{self.current_status.files} "
+                        f"files completed"
+                    )
+                    err_str = f"Request {titlestr}was canceled"
+                    if self.current_status.log_url is not None:
+                        err_str += f"\nLogfiles at {self.current_status.log_url}"
+                    raise ServiceXException(err_str)
```

Comment:
The code that generates the message here looks identical to above - and complex. Should this be refactored?

Reply:
I feel it differs enough between the two invocations that refactoring would be rather ugly. If we for some reason wind up using the pattern a third time I would probably reconsider.